saleor/dashboard/product/forms.py

```diff
@@ -209,10 +209,16 @@ def save(self, commit=True):
         attributes = {}
         for attr in self.product_attributes:
             value = self.cleaned_data.pop(attr.get_formfield_name())
-            if isinstance(value, AttributeChoiceValue):
-                attributes[smart_text(attr.pk)] = smart_text(value.pk)
-            else:
-                attributes[smart_text(attr.pk)] = value
+
+            # if the passed attribute value is a string,
+            # create the attribute value.
+            if not isinstance(value, AttributeChoiceValue):
```

Comment:
This block is the same as in `VariantAttributeForm`, maybe we can make a function from it?